lua/telescope/builtin/files.lua

```diff
@@ -62,10 +64,16 @@ end
 --  opts.search -- the string to search.
 --  opts.search_dirs -- list of directory to search in
 files.grep_string = function(opts)
-  -- TODO: This should probably check your visual selection as well, if you've got one
 
   local vimgrep_arguments = opts.vimgrep_arguments or conf.vimgrep_arguments
   local search_dirs = opts.search_dirs
+
+  -- set search from visual mode
+  if vim.tbl_contains(supported_visual_modes, vim.fn.mode()) then
+      opts.search = utils.get_visual_selection(opts.delimiter)
+      vim.api.nvim_command('normal :esc<CR>')
```

Comment:
Usually we write `vim.cmd [[...]]` . In this case I think maybe you want `:stopinsert` though?

Reply:
I see! `:stopinsert` throws an error though saying `no range allowed` since its called from visual mode. I'll recheck to see if there is a more idiomatic way to leave visual mode.